src/block/block_open.c

```diff
@@ -316,6 +316,16 @@ __desc_read(WT_SESSION_IMPL *session, uint32_t allocsize, WT_BLOCK *block)
     if (F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
         return (0);
 
+    /*
+     * If a data file is smaller than the allocation size, we're not going to be able to read the
+     * descriptor block. We should treat this as if the file has been deleted; that is, to log an
+     * error but continue on.
+     */
+    if (block->size < allocsize)
```

Comment:
Thanks @tetsuo-cpp this change looks OK to me. Do you understand why it's failing now, and it didn't used to fail with the same test case?

Reply:
This is failing in the rollback to stable path. Essentially that involves iterating over the metadata entries and performing a rollback to stable on each data file. This entire code path doesn't exist in 4.2.

Reply:
I guess I'm not seeing something. There would have been a path where we tried to open the file before, even if it wasn't during recovery. Right?
I'd like to talk to you in person, but I've got another meeting - so I won't get to for at least 30 minutes more.